include/tunnler/room_member.h

```diff
@@ -108,31 +112,41 @@ class RoomMember final {
      * Attempts to join a room at the specified address and port, using the specified nickname.
      * This may fail if the username is already taken.
      */
-    void Join(const std::string& nickname, const std::string& server = "127.0.0.1", const uint16_t serverPort = DefaultRoomPort, const uint16_t clientPort = 0);
+    void Join(const std::string& nickname, const std::string& server = "127.0.0.1",
+              const uint16_t serverPort = DefaultRoomPort, const uint16_t clientPort = 0);
 
     /**
      * Leaves the current room.
      */
     void Leave();
-    
+
 private:
-    State state;    // Current state of the RoomMember
-    MemberInformation member_information;
-    RoomInformation room_information;
+    State state; ///< Current state of the RoomMember.
+    MemberList member_information; ///< Information about the clients connected to the same room as us.
+    RoomInformation room_information; ///< Information about the room we're connected to.
 
-    RakNet::RakPeerInterface* peer;   // RakNet network interface
+    RakNet::RakPeerInterface* peer; ///< RakNet network interface.
 
     /**
      * Extracts a WifiPacket from a received RakNet packet and adds it to the proper queue.
      * @param packet The RakNet packet that was received.
      */
     void HandleWifiPackets(const RakNet::Packet* packet);
 
-    std::thread receive_thread; // Thread that receives and dispatches network packets
+    std::unique_ptr<std::thread> receive_thread; ///< Thread that receives and dispatches network packets
+
+    /// Max size of the beacon queue before the oldest entry is expunged.
+    /// This number was empirically calculated, keeping too many frames will
+    // cause an overflow in UDS::RecvBeaconBroadcastData.
+    static const size_t MaxBeaconQueueSize = 25;
+
+    std::deque<ChatEntry> chat_queue;    ///< List of all chat messages recieved since last PopChatEntries was called
+    std::deque<WifiPacket> data_queue;   ///< List of all recieve 802.11 frames with type Data
```

Comment:
spelling: "received" - Maybe also put the word "Data" in backticks / quotes?
Same with Beacon line below